fmriprep/workflows/anatomical.py

```diff
@@ -74,10 +76,130 @@ def t1w_preprocessing(name='t1w_preprocessing', settings=None):
         ),
         name='T1_2_MNI_Registration'
     )
-    # should not be necesssary byt does not hurt - make sure the multiproc
+    # should not be necesssary but does not hurt - make sure the multiproc
     # scheduler knows the resource limits
     t1_2_mni.interface.num_threads = settings['ants_nthreads']
 
+    # 6. FreeSurfer reconstruction
+    if settings['freesurfer']:
+        nthreads = settings['nthreads']
+
+        def detect_inputs(t1w_list, default_flags=''):
+            from nipype.utils.filemanip import filename_to_list
+            import nibabel as nib
+            t1w_list = filename_to_list(t1w_list)
+            t1w_ref = nib.load(t1w_list[0])
+            hires = max(t1w_ref.header.get_zooms()) < 1
+            t1w_outs = [t1w_list.pop(0)]
+            for t1w in t1w_list:
+                img = nib.load(t1w)
+                if all((img.shape == t1w_ref.shape,
+                        img.header.get_zooms() == t1w_ref.header.get_zooms())):
+                    t1w_outs.append(t1w_outs)
+
+            autorecon1_flags = [default_flags]
+            reconall_flags = [default_flags]
+            if hires:
+                autorecon1_flags.append('-hires')
+                reconall_flags.append('-hires')
+            return (t1w_outs, ' '.join(autorecon1_flags),
+                    ' '.join(reconall_flags))
+
+        recon_config = pe.Node(
+            niu.Function(
+                function=detect_inputs,
+                input_names=['t1w_list', 'default_flags'],
+                output_names=['t1w', 'autorecon1_flags', 'reconall_flags']),
+            name='ReconConfig',
+            run_without_submitting=True)
+        recon_config.inputs.default_flags = '-noskullstrip'
+
+        def bidsinfo(in_file):
+            from fmriprep.interfaces.bids import BIDS_NAME
+            match = BIDS_NAME.search(in_file)
+            params = match.groupdict() if match is not None else {}
+            return tuple(map(params.get, ['subject_id', 'ses_id', 'task_id',
+                                          'acq_id', 'rec_id', 'run_id']))
```

Comment:
This could be useful for pybids. I've opened an issue https://github.com/INCF/pybids/issues/31